libpod/container_internal_linux.go

```diff
@@ -49,10 +50,15 @@ func (c *Container) mountSHM(shmOptions string) error {
 	return nil
 }
 
-func (c *Container) unmountSHM(mount string) error {
+func (c *Container) unmountExtra(mount string) error {
+	if err := overlay.CleanupMount(mount); err != nil {
+		logrus.Warnf("container %s failed to cleanup overlay content %s : %v", c.ID(), mount, err)
+	}
 	if err := unix.Unmount(mount, 0); err != nil {
-		if err != syscall.EINVAL && err != syscall.ENOENT {
-			return errors.Wrapf(err, "error unmounting container %s SHM mount %s", c.ID(), mount)
+		if err != syscall.EINVAL {
```

Comment:
Why are we dropping errors here? I really don't like ignoring errors from unmount. They lead to unpleasant situations where we can't get rid of the container.

Reply:
No idea, added back.